### PR TITLE
[SELS-TASK] Integration of User Authentication with Axios

### PR DESCRIPTION
**Pull request by NiloJr-sun**
## Asana Link 
- [Integration of User Authentication with Axios](https://app.asana.com/0/1202703680910766/1202711099867213/f)
## Commands
```
- cd backend
- php artisan serve
- cd frontend
- npm start
```
##  Difinition of done
- [x] Integration of User Authentication

## Notes
- 

## Screenshots
image
image
image
image

### CODE REVIEW

File: backend/app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -34,30 +33,36 @@ public function create(Request $request)
...
         $field = $request->validate([
-            'email' => ['required'],
+            'name' => ['required'],
             'password' => ['required'],
         ]);
-        $res = User::where('email', $request->email)->first();
-        if (!$res || !Hash::check($field['password'], $res->password)) {
+        $res = User::where('name', $request->name)->first();
+        if ($res) {
+            if (!Hash::check($field['password'], $res->password)) {
```
Comment (ejanton-sun) [inline backend/app/Http/Controllers/UserController.php]:
```suggestion
            if ($res && !Hash::check($field['password'], $res->password)) {
```

File: frontend/src/components/api/axios.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export default axios.create({
+    baseURL: 'http://127.0.0.1:8000/api'
+})
```
Comment (ejanton-sun) [inline frontend/src/components/api/axios.js]: add new line

File: frontend/src/components/home/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+const Dashboard = () => {
+
+    return (
+
+        <h1>dashboard ni</h1>
+    )
+}
+
+export default Dashboard
```
Comment (ejanton-sun) [inline frontend/src/components/home/dashboard.js]: add new line

File: frontend/src/components/home/dashboard.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+
+const Dashboard = () => {
+
+    return (
+
```
Comment (ejanton-sun) [inline frontend/src/components/home/dashboard.js]: remove extra line
```suggestion
```

File: frontend/src/components/login/login.js [hunk 5]
```diff
@@ -20,42 +26,77 @@ const Login = () => {
...
+                });
+            setSuccess(true);
+
+        } catch (err) {
+            if (!err.response) {
+                setErrMsg('No server Response');
+            } else if (err.response?.status === 401) {
+                setErrMsg(err.response.data.message);
+            } else {
+                // setErrMsg('Registration failed');
```
Comment (ejanton-sun) [inline frontend/src/components/login/login.js]: remove comment if not used

File: frontend/src/index.css [hunk 6]
```diff
@@ -83,37 +82,37 @@ button {
...
     color: firebrick;
     font-weight: bold;
     padding: 0.5rem;
     margin-bottom: 0.5rem;
 }
 
-.line {
+.auth .line {
     display: inline-block;
 }
```
Comment (ejanton-sun) [inline frontend/src/index.css]: add new line

**Review by ejanton-sun — CHANGES_REQUESTED** [review]

File: backend/app/Http/Controllers/UserController.php (hunk 1 above)
Comment (NiloJr-sun) [reply backend/app/Http/Controllers/UserController.php]: Hello sir, regarding on this one. I needed the code to return specifically if the the username exists, and had it checked with the retrieved password given $res returns true. Should I just replace it with a "Wrong Credentials" as a whole?

**Review by ejanton-sun — APPROVED** [review]
# Approved

**Pull request merged**